docs/Using_Benchmark.md

```diff
@@ -0,0 +1,285 @@
+commands in the terminal.
+
+## In Windows
+```bash
+> cl /EHsc /MD BenchmarkEverything.cpp /I<BENCHMARK_SOURCE_DIR>\include /link shlwapi.lib <BENCHMARK_BUILD_DIR>\src\Release\benchmark.lib
+```
+where, `<BENCHMARK_SOURCE_DIR>` is the directory where Benchmark was cloned into
+and `<BENCHMARK_BUILD_DIR>` the directory where Benchmark was built. Here, we 
+had built Benchmark in the Release configuration and hence we used 
+`<BENCHMARK_BUILD_DIR>\src\Release\benchmark.lib`. We also used the `cl` 
+compiler which is provided with Visual Studio and also linked the `shlwapi` 
+library.
+
+## In Linux
+```bash
+$ clang++ -std=c++11 BenchmarkEverything.cpp -I<BENCHMARK_SOURCE_DIR>/include -pthread -L <BENCHMARK_BUILD_DIR>/src -lbenchmark
+```
+where `<BENCHMARK_SOURCE_DIR>` is the directory where Benchmark was cloned 
+into and `<BENCHMARK_BUILD_DIR>` the directory where Benchmark was built. 
+We also need to link the `pthread` library.
+
+# Embedding Benchmark in other projects
+ 
+Benchmark can easily be included in other projects. There are a few ways by 
+which this can be done,
+*   Clone the repository and build it inside the project and then use it 
+    in the project.
+*   Add Benchmark as a git submodule and incorporate it as a part of the 
+    project.
+*   Use CMake to download and build Benchmark as a part of the project 
+    and use it in the project.
+
+For the first method we can follow the usual steps to build and use Benchmark.
+The second and the third methods are described below
+
+
+## Adding Benchmark as a git submodule
+
+We can add Benchmark as a submodule in our CMake project, build it and use it 
+in the project. To demonstrate this method we consider a simple project 
+`BenchmarkEverything` whose directory structure is as follows:
+
+```bash
+BenchmarkEverything
+|   CMakeLists.txt
+|   BenchmarkEverything.cpp
+|   .git
+```
+
+We add Benchmark as a submodule to our git repository using
+```bash
+$ git submodule add http://github.com/google/benchmark.git extproject/benchmark
+```
+This will add Benchmark as a submodule in `/extproject/benchmark/`. In our 
+`CMakeLists.txt` file we incorporate this submodule into our project using `add_subdirectory()`
+```cmake
+# setting minimum cmake version
+cmake_minimum_required (VERSION 3.14)
+# setting cxx standard
+set (CMAKE_CXX_STANDARD 11)
+set (CMAKE_CXX_STANDARD_REQUIRED ON)
+# project name
+project (BenchmarkEverything)
+# disabling tests for benchmark
+set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Disable benchmark tests" FORCE)
+add_subdirectory(extprojects/benchmark)
+# including benchmark header files
+include_directories(extprojects/benchmark/include)
+add_executable(BenchmarkEverything BenchmarkEverything.cpp)
+target_link_libraries(BenchmarkEverything benchmark)
+if(WIN32)
+    target_link_libraries(BenchmarkEverything shlwapi)
+elseif(LINUX)
+    target_link_libraries(BenchmarkEverything pthread)
```

Comment:
This is a bad advise.
The library needs to be found first.

Reply:
I tried to keep it simple if the library can't be found the error will say that

Reply:
link-time error
link-time errors are not nice :/